File: Garage_Rental.Infra/Service/RentService.cs
using Garage_Rental.Core.Data;
using Garage_Rental.Core.Repository;
using Garage_Rental.Core.Service;
using System;
using System.Collections.Generic;
using System.Text;

namespace Garage_Rental.Infra.Service
{
    public class RentService : IGenericService<Rent>
    {
        private readonly IGenericRepository<Rent> _rentRepository;

        public RentService(IGenericRepository<Rent> RentRepository)
        {
            _rentRepository = RentRepository;
        }

        public bool Create(Rent t)
        {
            return _rentRepository.Create(t);
        }

        public void Delete(int id)
        {
            _rentRepository.Delete(id);
        }

        public List<Rent> GetAll()
        {
            return _rentRepository.GetAll();
        }

        public Rent GetById(int id)
        {
            return _rentRepository.GetById(id);
        }

        public void Update(Rent t)
        {
            _rentRepository.Update(t);
        }
    }
}

File: FrontEnd/GarageRentalFrontEnd/src/app/Services/rent.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { Rent } from '../Models/rent.model';

@Injectable({
  providedIn: 'root'
})
export class RentService {


 baseApiUrl:string = environment.baseApiUrl;

  constructor(private http:HttpClient) { }


  getRentList(): Observable<Rent[]>{
    return this.http.get<Rent[]>(this.baseApiUrl +'Rent/GetAll');
  }

  getRent(id:string):Observable<Rent>{
    return this.http.get<Rent>(this.baseApiUrl +'Rent/GetById?id=' + id);
  }

  addRent(addRentReq:Rent):Observable<Rent>{
    return this.http.post<Rent>(this.baseApiUrl +'Rent/Create',addRentReq);
  }

  updateRent(id:number,updateRentreq:Rent):Observable<Rent>{
    return this.http.post<Rent>(this.baseApiUrl +'Rent/Update?id='+ id ,updateRentreq)
  }
  deleteRent(id:number):Observable<Rent>{
    return this.http.delete<Rent>(this.baseApiUrl +'Rent/Delete?id=' + id);
  }
}
